lib/asciidoctor/substitutors.rb

```diff
@@ -542,14 +542,17 @@ def sub_macros text
         end
 
         prefix, suffix = $1, ''
-        # NOTE if $4 is set, then we're looking at a formal macro
+        # NOTE if $4 is set, we're looking at a formal macro (e.g., https://example.org[])
         if $4
           prefix = '' if prefix == 'link:'
           text = $4
         else
-          # invalid macro syntax (link: prefix w/o trailing square brackets)
-          # FIXME we probably shouldn't even get here...our regex is doing too much
-          next $& if prefix == 'link:'
+          # invalid macro syntax (link: prefix w/o trailing square brackets or enclosed in double quotes)
+          # FIXME we probably shouldn't even get here when the link: prefix is present; the regex is doing too much
+          case prefix
+          when 'link:', '"', ?'
```

Comment:
nitpick, the last token `?'` is equivalent to `"'"` or `'\''` right?
I'm not against this notation but maybe we should use it consistently:

```rb
when 'link:', ?", ?'
```

or, if we want to use single quotes everywhere:

```rb
when 'link:', '"', '\''
```

Reply:
I do want to switch to it everywhere, but that's for a separate refactoring. For now, I will make these two consistent (and forward looking).